test/unit/src/common/selectors/t_selectors.js

```diff
@@ -0,0 +1,37 @@
+import expect from 'expect';
+
+import { hasNoRegisteredNames } from '../../../../../src/common/selectors';
+
+describe('hasNoRegisteredNames', function() {
+
+  const stateWithNoName = {
+    snaps: {
+      snaps: [{}, {}]
+    }
+  };
+
+  const stateWithName = {
+    snaps: {
+      snaps: [{
+        store_name: 'bsi-test-ii'
+      }, {}]
+    }
+  };
+
+  it('should be false when no names in store', function() {
+    expect(hasNoRegisteredNames(stateWithNoName)).toBe(true);
+  });
+
+  it('should be true when name in snaps', function() {
+    expect(hasNoRegisteredNames(stateWithName)).toBe(false);
+  });
+
+  it('should not memoize the selector', function() {
+    expect(hasNoRegisteredNames.recomputations()).toBe(2);
```

Comment:
Yes, but that's intended. I'll wrap it up into it's own describe block and use `resetRecomputations` to make that clear.